src/verifier/step3/Step3Data.sol

```diff
@@ -0,0 +1,139 @@
+        uint256[] memory r_U_X = new uint256[](2);
+        // Limbs order should be reversed and every limb should be reversed for each r_U_X
+        r_U_X[0] = 0x1afccc8b4eaa3bc1718194359f53d11f56c9dcbc44b435dd8acd4055542bf541;
+        r_U_X[1] = 0x0b3f6387ca3de17a52ca09ab0ec108a3c793cb930f745f5f788507aa4e5e18e9;
+        uint256 r_U_u = Field.reverse256(0xebb5df91f82f07ccef5a3f3b56546bcc00000000000000000000000000000000);
+        uint256 l_u_comm_W = 0x73cb2576d50f1f5e74dce3cf18a48deb5ee483e870d5e8e8a8605b07dfe1920c;
+        uint256[] memory l_u_X = new uint256[](2);
+        l_u_X[0] = Field.reverse256(0xca7562fbd1521f53030240a9ebc049769c81aa100d3c3d4b4aad6cdd381dd402);
+        l_u_X[1] = Field.reverse256(0x324a23526a206e6ea1077a87c58b61d1a9207365802bc67b9caf559f03218203);
+        return (
+            Field.reverse256(0xdca18ed3cacabc2daf70380c287c0840b694735dcc63375d49912a5676d5f300),
+            nifs,
+            r_U_comm_W,
+            r_U_comm_E,
+            r_U_X,
+            r_U_u,
+            l_u_comm_W,
+            l_u_X
+        );
+    }
+
+    /* This function returns the following results in order:
+    * pp_digest, NIFS instance, r_U comm_W, r_U comm_E, r_U X, r_U u
+    * l_u comm_W, l_u X
+    */
+    function returnSecondaryData()
+        public
+        pure
+        returns (uint256, bytes32, uint256, uint256, uint256[] memory, uint256, uint256, uint256[] memory)
+    {
+        uint8[32] memory nifs_array;
+        nifs_array[0] = 51;
+        nifs_array[1] = 28;
+        nifs_array[2] = 184;
+        nifs_array[3] = 191;
+        nifs_array[4] = 145;
+        nifs_array[5] = 149;
+        nifs_array[6] = 208;
+        nifs_array[7] = 155;
+        nifs_array[8] = 200;
+        nifs_array[9] = 88;
+        nifs_array[10] = 188;
+        nifs_array[11] = 40;
+        nifs_array[12] = 1;
+        nifs_array[13] = 24;
+        nifs_array[14] = 67;
+        nifs_array[15] = 125;
+        nifs_array[16] = 209;
+        nifs_array[17] = 153;
+        nifs_array[18] = 46;
+        nifs_array[19] = 23;
+        nifs_array[20] = 49;
+        nifs_array[21] = 243;
+        nifs_array[22] = 244;
+        nifs_array[23] = 155;
+        nifs_array[24] = 67;
+        nifs_array[25] = 33;
+        nifs_array[26] = 24;
+        nifs_array[27] = 58;
+        nifs_array[28] = 215;
+        nifs_array[29] = 175;
+        nifs_array[30] = 103;
+        nifs_array[31] = 6;
+        bytes32 nifs = Field.uint8ArrayToBytes32(nifs_array);
+        uint256 r_U_comm_W = 0xfeb641951cb4c21bdc4906f57b46942c0ae22b6c852266f64a4e0b77de5c4432;
+        uint256 r_U_comm_E = 0x8412865c490c9e1219066e4e257f3bb28e00db31736251586c86197f6ec866a2;
+        uint256[] memory r_U_X = new uint256[](2);
+        // Limbs order should be reversed and every limb should be reversed for each r_U_X
+        r_U_X[0] = 0x107b0ae1735bf84de833f1d79b14709113a2e59ba6b3e153d39a7ed25e72441a;
```

Comment:
I think this value was "crafted" manually and I'm not sure, if Python script produces this exact version of `Step3Data.sol` contract.